docs: changes to BUILD.md and CONTRIBUTING.md #1792

ruzell22 opened the pull request:
Document updates for issue #1792

ruzell22 commented:
Hi @petermetz this is the ticket for the document update issue I have created the other day. Thank you.

petermetz commented on BUILD.md:
```diff
@@ -89,7 +89,8 @@ The `npm run watch` script in action:
   * Git
-  * NodeJS 16, npm 7 (we recommend using the Node Version Manager (nvm) if available for your OS)
+  * NodeJS v16.13.1, npm v8.1.2 (we recommend using the Node Version Manager (nvm) if available for your OS)
+    * `nvm use 16.13.1`
```
@ruzell22 I would say `nvm install 16.13.1` instead of `nvm use ...` because if you are on a fresh installation of nvm, it won't let you `use` a specific version unless it was `install` 'd first (at least this is how it used to work a couple of years back the last time I used it, please do correct me if they changed the behavior in the meantime though)

ruzell22 commented:
@petermetz I have checked and verified that nvm install is to be run first before nvm use to ensure that the user is using node v16.13.1. Thank you for that. 

I have fixed it on the PR and also cleaned the commit numbers in this ticket. Let me know if there is any more changes needed. Thank you.

ruzell22 commented:
@petermetz added changes on prerequisites, openJDK. Thank you

petermetz commented on BUILD.md:
```diff
@@ -89,11 +89,17 @@ The `npm run watch` script in action:
   * Docker Engine
   * Docker Compose
+  * OpenJDK
+    * `sudo apt install openjdk-17-jre-headless`
```
@ruzell22 This would install the Java Runtime Environment not the Java Development Kit which is not enough unless you are an end user running a Java based application. I think the one you want is probably `openjdk-11-jdk` or maybe `openjdk-11-jdk-headless` 

The reason I'm saying 11 instead of 17 is because I'm not sure if 17 can build the Corda/Springboot code components. If you could verify that it works with `openjdk-17-jdk` then I'd be fine bumping it up to 17 as well so just let me know.

petermetz reviewed (changes requested):
@ruzell22 When you have several commits with the same exact message it is recommended to do `fixup` instead of `squash` when interactively rebasing so that the commit message does not get repeated needlessly. I marked the redundant parts on this screenshot:

Also: The `Fixes #????` syntax should also be in the commit message body, apologies for not mentioning this earlier.

![Screenshot from 2022-01-26 16-21-58](https://user-images.githubusercontent.com/1583231/151269024-0a4ea358-caef-491f-9da1-95124fcb295e.png)

ruzell22 commented:
@petermetz According to this link https://docs.r3.com/en/platform/corda/4.7/open-source/getting-set-up.html , r3 corda requires JDK 8 and currently does not support 9 or higher, hence the openjdk-8-jdk-headless in the latest push in the PR.

As for the commit message, I have fixed it into one commit title and with message 'fixes: #????'. Thank you

ruzell22 commented:
Apologies @petermetz , I was pushing a new change adding the reason in the BUILD.md file as to why it was decided to be OpenJDK 8. Let me put a screenshot
![image](https://user-images.githubusercontent.com/89818529/151498503-5db62811-31bf-45e9-bb6c-ed2da33692bf.png)
However, the ticket suddenly closed when I forced push it with lease to reflected the new change to ask for review points.

What do you suggest I do? Apologies for accidentally closing the ticket.

petermetz commented:
> What do you suggest I do? Apologies for accidentally closing the ticket.

@ruzell22 Let's get back to the review once you've restored the PR diff. What I recommend you do is find the local branch where you do have your changes still and then put them back (either manually or via cherry pick) onto whatever branch you have mapped locally to your remote branch [1]. You can likely do so [2] even if you deleted the branch itself or if you amended it to not have the previously made commits at all (which is what it looks like you did).

[1] https://github.com/ruzell22/cactus/tree/docschangestobuildandcontributingmd
[2] https://stackoverflow.com/a/10099285
